public/components/overview/components/dashboard_controls.tsx

```diff
@@ -13,19 +14,56 @@ import {
   EuiToolTip,
 } from '@elastic/eui';
 import { OnTimeChangeProps } from '@opensearch-project/oui/src/eui_components/date_picker/super_date_picker/super_date_picker';
-import React from 'react';
 import { useObservable } from 'react-use';
 import { FormattedMessage } from '@osd/i18n/react';
 import { coreRefs } from '../../../framework/core_refs';
 import { HOME_CONTENT_AREAS } from '../../../plugin_helpers/plugin_overview';
 import { redirectToDashboards } from '../../getting_started/components/utils';
 import { AddDashboardCallout } from './add_dashboard_callout';
+import { AddDataSourceCallout } from './add_datasource_callout';
 import { ObsDashboardStateManager } from './obs_dashboard_state_manager';
+import { SavedObjectsClientCommonFindArgs } from '../../../../../../src/plugins/data/common';
+import { getWorkspaceIdFromUrl } from '../../../../../../src/core/public/utils';
+import { setObservabilityDashboardsId } from './utils';
+
+const getDatasourceAttributes = async () => {
+  const findOptions: SavedObjectsClientCommonFindArgs = {
+    type: 'data-source',
+    perPage: 1000,
+  };
+
+  const allDataSources = await coreRefs?.savedObjectsClient?.find(findOptions);
+  return allDataSources?.savedObjects ?? [];
+};
 
 export function DashboardControls() {
+  const [isDataSourceEmpty, setIsDataSourceEmpty] = useState<boolean | null>(null);
+  const [isInWorkspace, setIsInWorkspace] = useState<boolean>(false); // Track if user is in a workspace
   const isDashboardSelected = useObservable(ObsDashboardStateManager.isDashboardSelected$);
   const dashboardState = useObservable(ObsDashboardStateManager.dashboardState$);
 
+  useEffect(() => {
+    const checkDataSource = async () => {
+      const currentUrl = window.location.href;
+      const workspaceId = getWorkspaceIdFromUrl(currentUrl, coreRefs?.http?.basePath.getBasePath());
+
+      if (workspaceId) {
+        setIsInWorkspace(true);
+        const savedObjectsArray = await getDatasourceAttributes();
+        setIsDataSourceEmpty(savedObjectsArray.length === 0);
+
+        // Set to null if there are no data sources associated [Handle if dashboard was set, then datasource deleted]
+        if (savedObjectsArray.length === 0) {
+          await setObservabilityDashboardsId(null);
```

Comment:
For the refresh problem, I looked into the CMS implementation. It looks like this is an issue from the CMS plugin where the registration is only happening in plugin setup: `https://github.com/opensearch-project/dashboards-observability/blob/6a43f539745ad1d51d66d99962563c620b681e4b/public/plugin.tsx#L192` May be we can add a call-back for this function. This callback can be called to run `setupOverviewPage`

Reply:
<img width="1863" alt="Screenshot 2024-11-11 at 10 46 31 AM" src="https://github.com/user-attachments/assets/dd952ea3-cbb5-4262-bf5b-7c20fe1aa328">
Wasn't able to use a callback as the registration exist. Found a work around calling
 
getOverviewPage().removeSection(SECTIONS.DASHBOARD);

to remove the section getting it to re-render correctly.